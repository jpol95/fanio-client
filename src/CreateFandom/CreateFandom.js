import FandomForm from "../FandomForm/FandomForm";
import React from "react";
import FetchService from "../FetchService";
import FanioContext from "../FanioContext";

export default class CreateFandom extends React.Component {
  userId = this.props.match.params.userId;
  static contextType = FanioContext;
  handleSubmit = async (e, fandom) => {
    e.preventDefault();
    const fandomResult = await FetchService.postFandom({
      ...fandom,
      userId: this.userId,
    });
    this.props.history.push(
      `/users/${this.userId}/fandom-view/${fandomResult.id}`
    );
    this.context.handleAddFandom(fandomResult);
  };

  render() {
    const startInfo = { title: "" };
    return (
      <>
        <h3>Create Fandom</h3>
        <FandomForm {...startInfo} handleSubmit={this.handleSubmit} />
      </>
    );
  }
}
