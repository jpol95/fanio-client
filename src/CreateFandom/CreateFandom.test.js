import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CreateFandom from './CreateFandom';

describe('Create fandom component', () => {
    const props ={
    match: {
        params: {
            userId: 1
        }
    }
}
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CreateFandom {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<CreateFandom {...props}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});