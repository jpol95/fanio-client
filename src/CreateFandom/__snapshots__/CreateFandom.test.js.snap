// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages component renders the UI as expected 1`] = `
Array [
  <h3>
    Create Fandom
  </h3>,
  <div
    className="fandom"
  >
    <form
      className="create-fandom"
      onSubmit={[Function]}
    >
      <label
        htmlFor="workname"
      >
        Name of fandom*
      </label>
      <input
        defaultValue=""
        id="workname"
        onChange={[Function]}
        type="text"
      />
      <button
        className="submit-fandom-button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
]
`;
